product_gtin/product_gtin.py

```diff
@@ -37,6 +37,53 @@ def check_ean(eancode):
         int(eancode)
     except:
         return False
+    check = True
+    if len(eancode) == 8:
+        check = check_ean8(eancode)
+    if len(eancode) == 11:
+        check = check_ean11(eancode)
+    if len(eancode) == 12:
+        check = check_upc(eancode)
+    if len(eancode) == 13:
+        check = check_ean13(eancode)
+    if len(eancode) == 14:
+        check = check_gtin14(eancode)
+    return check
```

Comment:
Or better:
```
check_func = getattr(self, 'check_ean_%s' %len(eancode), None)
if check_func:
    return check_func(eancode)
return False
```
And you will have to make homogeneus all method names (check_ean or check_gtin, what you want).